enhancements/certificate-short-rotation.md

```diff
@@ -0,0 +1,132 @@
+---
+title: certificate-short-rotation
+authors:
+  - vrutkovs
+reviewers:
+  - deads2k
+approvers:
+  - deads2k
+api-approvers:
+  - deads2k
+creation-date: 2024-08-24
+last-updated: 2024-08-24
+tracking-link:
+  - https://issues.redhat.com/browse/API-1688
+---
+
+# Short Rotation Period For Certificates
+
+## Summary
+
+Add new feature gate in DevPreview set so that components would issue certificates with shorter 
+duration - hours instead of days.
+
+## Motivation
+
+Currently certificates are issued by Openshift with various validity durations, but at least its 15 
+days. This makes testing certificate rotation in CI complicated - we have to emulate passing time 
+using time skewing. This methods shows how cluster recovers after certificates have expired, but 
+it doesn't help us with testing happy path when certificates rotate during standard cluster lifecycle.
+
+Some components (i.e. cluster-kube-apiserver-operator) issue certificate with shorter lifetime in 
+development branch. This requires us to revert this change every time we branch for new release.
+This also doesn't help us in CI, as it needs a similar change in the installer. 
+Also, most components are not using this, so we end up with some certificates valid for hours but  
+most would be valid for days.
+
+Since the change to revert this setting requires manual pull request, there is chance that this 
+setting will leak into supported releases.
+
+This enhancement describes a new feature gate, which would enable this feature for all components 
+and ensure that stable releases don't have it accidentally enabled as it uses FeatureGates.
+
+### User Stories
+
+> As an Openshift developer, I want to have a setting for component to issue shorter living 
+> certificates so that I could verify that certificate rotation doesn't cause issues
+
+### Goals
+
+* Create a new FeatureGate in DevPreview featureset
```

Comment:
>a) can a customer enable this feature on their cluster?

As any other featuregate - yes, as any other featuregate from DevPreviewNoUpgrade it would render the cluster unsupported and unupgradable

> have we thought about of something static, that can not be easily manipulated by a customer, like a build variable infused in CI so all components build with it.

We have a `unsupported-cert-rotation-config` configmap used by cluster-kube-apiserver-operator, which changes only `Upgradable` condition for kube-apiserver. The issue with this method is that its not coordinated with other components - most notably installer, so its not helpful in CI where we want to have all certificates regenerated in 4-8 hours without manual intervention.

Reply:
> As any other featuregate - yes, as any other featuregate from DevPreviewNoUpgrade it would render the cluster unsupported and unupgradable

okay, so a customer can enable the feature, and the cluster would be marked as "Not Upgradeable". But what level of guarantee/confidence are we providing that it won't break the cluster? The customer would still lodge a complaint with us if the cluster degrades, right? 
I think with `DevPreviewNoUpgrade`, we say no upgarde, but we still provide some level of guarantee in terms of cluster functioning properly. Thoughts?

Reply:
>what level of guarantee/confidence are we providing that it won't break the cluster?

Same guarantees apply - my only concern is availability due to cert rotation rollouts. We can guarantee that with e2e tests

>The customer would still lodge a complaint with us if the cluster degrades, right?

They can file a support ticket, but DevPreview clusters are not supportable without an exception